src/core/qgsthreadingutils.h

```diff
@@ -43,6 +47,14 @@
 #define QGIS_PROTECT_QOBJECT_THREAD_ACCESS_NON_FATAL do {} while(false);
 #endif
 
+#ifdef AGGRESSIVE_SAFE_MODE
+#define QGIS_CHECK_QOBJECT_THREAD_EQUALITY(other) if ( other->thread() != thread() ) {qFatal( "%s", QStringLiteral("%2 (%1:%3) Object %4 is from a different thread than the object %5 lives in [0x%6 vs 0x%7]" ).arg( QString( __FILE__ ), QString( __FUNCTION__ ), QString::number( __LINE__  ), other->objectName(), objectName() ).arg( reinterpret_cast< qint64 >( QThread::currentThread() ), 0, 16 ).arg( reinterpret_cast< qint64 >( thread() ), 0, 16 ).toLocal8Bit().constData() ); }
```

Comment:
That should probably be QDtringLiteral?

Reply:
Where?

Reply:
QString( __FILE__ ) etc?

Reply:
I'd be inclined not to -- the strings won't be constructed **unless** there's an issue (which is hopefully unlikely 🤪), so I'd take the miniscule improvement in compilation time over the miniscule cost in run time here!

Reply:
Then the same should apply for the message template itself (`"%2 (%1%3...`), no
FTR our time on this discussion is most certainly not worth the potential costs / benefits anyway ;-)

Reply:
Yeah. Honestly I just don't feel like waiting for this to complete another round of ci builds 😭

Reply:
CI build time is an issue that would deserve some attention indeed ...

Reply:
Discussed this yesterday with @3nids 
We assumed that we are using the cache too heavy (different jobs, os, artefacts from release and pr) and that results in too much pressure in the IIRC 10GB available.

We either need to review what we cache and for which branches ...
Or store caches somewhere where we have more storage

Reply:
Ok , QStringLiteral does NOT work here. Now I'm ready to smash my computer and rage quit this whole industry :scream:

Reply:
![giphy](https://user-images.githubusercontent.com/1728657/207537589-9aee2b5c-728c-4593-a44d-5a1e6e042c0b.gif)

Reply:
Bummer :angry: